ksql-engine/src/main/java/io/confluent/ksql/planner/plan/AggregateNode.java

```diff
@@ -271,12 +280,14 @@ protected int getPartitions(KafkaTopicClient kafkaTopicClient) {
 
   private Map<Integer, Integer> createAggregateValueToValueColumnMap(
       final SchemaKStream aggregateArgExpanded,
-      final SchemaBuilder aggregateSchema
+      final SchemaBuilder aggregateSchema,
+      final InternalSchema internalSchema
   ) {
     Map<Integer, Integer> aggValToValColumnMap = new HashMap<>();
     int nonAggColumnIndex = 0;
     for (Expression expression : getRequiredColumnList()) {
-      String exprStr = expression.toString();
+      String exprStr =
+          internalSchema.getExpressionToInternalColumnNameMap().get(expression.toString());
```

Comment:
Might be better to have a method like `InternalSchema.getInternalColumn(String expression)`. Encapsulates the `Map` (which is an implementation detail) a bit better.